src/engine/ScreenShotTaker.cs

```diff
@@ -26,18 +37,25 @@ public override void _Ready()
     public void TakeScreenshotPressed()
     {
         GD.Print("Taking a screenshot");
-        TakeAndSaveScreenShot();
+        TakeScreenshot();
     }
 
     /// <summary>
-    ///   Takes and saves a screenshot
+    ///   Takes an image of the current viewport
     /// </summary>
-    /// <returns>The path of the screenshot on success, null on failure</returns>
-    public string TakeAndSaveScreenShot()
+    /// <returns>The image</returns>
+    public Image GetViewportTextureImage()
```

Comment:
```suggestion
    public Image GetViewportTextureAsImage()
```
Or if you prefer shorter names I think `GetViewportAsImage` would be perfectly descriptive as well.